lib/katello/tasks/unify_hosts.rake

```diff
@@ -14,7 +16,12 @@ namespace :katello do
     end
 
     def unify_all
+      passed = []
       ::Host.unscoped.where("name like '%.%'").find_each do |host|
+        next if passed.include?(host.name.downcase)
+        passed << host.name.downcase
+        puts "Skipping #{host.name}, more than one record." && next if ::Host.unscoped.where(:name => host.name).count > 1
+
```

Comment:
OPTIONAL: can you add logic here to skip virt who. Optional because virt-who tends not to have host names so gets skipped anyway.